File: zero-rule-server-main/domain/src/main/java/server/domain/entity/ItemMgmt.java
package server.domain.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;
import server.domain.vo.UsedRuleInfoVo;

import java.time.Instant;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ItemMgmt {

    /**
     * 항목ID
     * PK
     */
    @NonNull
    private String itemid;

    /**
     * 항목명
     */
    private String itemNm;

    /**
     * 항목별칭명
     */
    private String itemAliasNm;

    /**
     * 항목설명내용
     */
    private String itemExplanDesc;

    /**
     * 항목데이터타입코드
     */
    private String dataTypeCd;

    /**
     * 항목데이터타입코드명
     */
    private String dataTypeNm;

    /**
     * 변경사용자ID
     */
    private String updateUserID;

    /**
     * 삭제여부(사용여부)
     */
    private String itemUseYn;

    /**
     * 최초등록사용자ID
     */
    private String firstRegUserId;

    /**
     * 변경일시
     */
    private String updateDateTime;

    /**
     * 최초등록일시
     */
    private String firstRegDateTime;

    /**
     * 인터페이스 아이디
     */
    private String ifid;

    private int usedCnt;




}

File: zero-rule-web/sub/domain/src/model/ItemMgmt.ts
export interface ItemMgmtUsedRuleInfo {
  ruleid: string;
  ruleNm: string;
  ruleState: string;
  activateYn: string;
}
/**
 * @interface ItemMgmt
 *
 * 항목관리 Select
 */
export interface ItemMgmt {
  itemid: string;
  itemNm?: string;
  itemAliasNm?: string;
  itemExplanDesc?: string;
  dataTypeCd?: string;
  dataTypeNm?: string;
  updateUserID?: string;
  updateDateTime?: string;
  itemUseYn?: string;
  firstRegUserId?: string;
  firstRegDateTime?: string;
  uptGubun?: string; // 룰관리 수정구분 화면에서 적용
  ifid?: string;
  usedCnt?: number;
}

export interface ItemRef {
  itemid: string;
  itemrefCd: string;
  itemrefNm: string;
  itemrefaliasNm: string;
  itemrefexprDesc: string;
  updateUserid: string;
  updateDatetime: string;
}
